.github/workflows/compilation.yml

```diff
@@ -33,3 +33,26 @@ jobs:
         export PS2DEV=$PWD/ps2dev
         export PATH=$PATH:$PS2DEV/dvp/bin
         ./toolchain.sh
+
+  build-win:
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install MSYS2 texinfo bison flex
```

Comment:
Can we do with the windows workflow something similar as we did with MacOS and Linux? 
having just `if: matrix.os == 'windows-latest'` (not sure if it is possible, just asking)

Reply:
Yes, forgot to mention why I didn't make it in such a way. windows target uses a different shell (msys2 shell not bash shell) for it, I didn't found a way how to change the shell based on matrix level.

Reply:
You can also do it using the matrix.
Now instead of define in the matrix the OS you can define the config, where you also will include the kind of shell to use, so it could be something as:

```yml
build:
    runs-on: ${{ matrix.config[0] }}
    defaults:
      run:
        shell: ${{ matrix.config[1] }}
    strategy:
      matrix:
        config: [[macos-latest, bash], [ubuntu-latest, bash], [windows-latest, msys2 {0}]]
```